sql/core/src/test/scala/org/apache/spark/sql/DataFrameSetOperationsSuite.scala

```diff
@@ -1018,6 +1018,23 @@ class DataFrameSetOperationsSuite extends QueryTest with SharedSparkSession {
     unionDF = df1.unionByName(df2)
     checkAnswer(unionDF, expected)
   }
+
+  test("SPARK-36673: Incorrect Unions of struct") {
```

Comment:
Could you be more specific if this aims to `nullability` issue?

Reply:
From the JIRA description, this is not with incorrect `nullability` but incorrect merged schema for union.

For example, the added test case should have only one column in the nested struct after union, but currently we got 2 columns.

Reply:
And current test title is too vague. Let me be specific here.